test/integration/scheduler/queue_test.go

```diff
@@ -177,7 +177,95 @@ func TestSchedulingGates(t *testing.T) {
 
 // TestCoreResourceEnqueue verify Pods failed by in-tree default plugins can be
 // moved properly upon their registered events.
-func TestCoreResourceEnqueue(t *testing.T) {
+func TestCoreResourceEnqueueWithQueueHint(t *testing.T) {
```

Comment:
We can reuse this function for #122305.
But I tend to leave this as a follow up, not refactor it in this pr.

Reply:
Does `TestCoreResourceEnqueueWithQueueHint` test the same thing (except the feature gate value) as `TestCoreResourceEnqueueWithOutQueueHint`? 
If Yes, can we make it something like:

```go
func TestCoreResourceEnqueue(t) {
	for _, qHintEnabled := range []bool{true, false} {
		t.Run(tt.name, func(t *testing.T) {
			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.SchedulerQueueingHints, qHintEnabled)()

			// write test...
		})
	}   
}
```